mitiq/zne/zne.py

```diff
@@ -28,7 +28,7 @@ def execute_with_zne(
     factory: Optional[Factory] = None,
     scale_noise: Callable[[QPROGRAM, float], QPROGRAM] = fold_gates_at_random,
     num_to_average: int = 1,
-) -> float:
+) -> Optional[float]:
```

Comment:
The fact that `execute_with_zne` returns an `Optional[float]` instead of a `float` is a bit confusing (this propagates also to zne decorators). But I guess this is the only way to make mypy happy. 

Would something like this be useful to fix the problem:

```python
zne_limit = factory.run(...).reduce()
if zne_limit is None:
    # I actually don't know how to generate this error but it could help mypy
    raise RuntimeError("Zero-noise extrapolation failed.")
return zne_limit
```

Otherwise I am fine with `Optional[float]`

Reply:
I don't know why it was optional actually. mypy is fine with just float. Fixed.